reference/ToDo/src/ToDo/App.xaml.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            return new LocalizableMessageDialogViewMap
+            (
+                Content: localizer => localizer![ResourceKey(ResourceKeys.DialogContent)],
+                Title: localizer => localizer![ResourceKey(ResourceKeys.DialogTitle)],
+                DelayUserInput: delayUserInput,
+                DefaultButtonIndex: defaultButtonIndex,
+                Buttons: buttons
+                    .Select(x => new LocalizableDialogAction(LabelProvider: localizer => localizer![ResourceKey(x.labelKeyPath)], Id: x.Id))
+                    .ToArray()
+            );
+            string ResourceKey(string keyPath)
+            {
+                // map absolute/relative path accordingly
+                return keyPath.StartsWith("./") ? keyPath.Substring(2) : $"Dialog_{section}_{keyPath}";
+            }
+        }
+
+        var deleteButton = (DialogResults.Affirmative, ResourceKeys.DeleteButton);
+        var cancelButton = (DialogResults.Negative, ResourceKeys.CancelButton);
+        var confirmDeleteListDialog = BuildDialogViewMap(Dialog.ConfirmDeleteList, true, 0, deleteButton, cancelButton);
+        var confirmDeleteTaskDialog = BuildDialogViewMap(Dialog.ConfirmDeleteTask, true, 0, deleteButton, cancelButton);
+        var confirmSignOutDialog = BuildDialogViewMap(Dialog.ConfirmSignOut, true, 0, (DialogResults.Affirmative, ResourceKeys.SignOutButton), cancelButton);
+
+        views.Register(
+            // Dialogs and Flyouts
+            new ViewMap<AddTaskFlyout, AddTaskViewModel>(),
+            new ViewMap<AddListFlyout, AddListViewModel>(),
+            new ViewMap<ExpirationDateFlyout, ExpirationDateViewModel>(Data: new DataMap<PickedDate>()),
+            new ViewMap<RenameListFlyout, RenameListViewModel>(),
+
+            // Views
+            new ViewMap<HomePage, HomeViewModel>(),
+            new ViewMap<TaskSearchFlyout>(),
+            new ViewMap<SearchPage, SearchViewModel>(),
+            new ViewMap<SettingsFlyout, SettingsViewModel>(),
+            new ViewMap<Shell, ShellViewModel>(),
+            new ViewMap<WelcomePage, WelcomeViewModel>(),
+            new DataViewMap<TaskListPage, TaskListViewModel, TaskList>(),
+            new DataViewMap<TaskPage, TaskViewModel, ToDoTask>(),
+            confirmDeleteListDialog,
+            confirmDeleteTaskDialog,
+            confirmSignOutDialog
+        );
+
+        routes.Register(
+            new RouteMap("", View: views.FindByViewModel<ShellViewModel>(), Nested: new RouteMap[]
+            {
+                new("Welcome", View: views.FindByViewModel<WelcomeViewModel>()),
+                new("Home", View: views.FindByViewModel<HomeViewModel>()),
+                new("TaskList", View: views.FindByViewModel<TaskListViewModel>(), Nested: new[]
+                {
+                    new RouteMap("ToDo", IsDefault:true),
+                    new RouteMap("Completed")
+                }),
```

Comment:
I think this is somehow related to the iOS navigation bug mentioned previously. Shouldn't TaskList be nested under Home? Since Home contains the NavigationView and then we are navigating within a Grid as the Content of the NavigationView.

I also see this which which may contribute to the problem: https://github.com/unoplatform/Uno.Samples/blob/723ed8c3b7f9c8a6242585f55a035642079121ad/reference/ToDo/src/ToDo/Views/HomePage.xaml#L40-L56

We seem to be changing the way we navigate to routes depending on the screen size.

@nickrandolph do you see anything here that we should fix? The problem I am seeing is on iOS when I click the back button on the NavigationBar in the TaskList page, it navigates back to Home with the NavigationView displayed but then automatically navigates forward to the TaskList page again

Reply:
@kazo0 we will open a separated issue for the iOS issue since it is also happening on master as well (before the migration)